packages/graphiql-build-fetcher/README.md

```diff
@@ -0,0 +1,147 @@
+
+### Setup
+
+[`graphiql`](https://npmjs.com/package/graphiql) and thus `react` and `react-dom` should already be installed.
+
+you'll need to install `@graphiql/build-fetcher`
+
+npm
+
+```bash
+npm install --save @graphiql/build-fetcher
+```
+
+yarn
+
+```bash
+yarn add @graphiql/build-fetcher
+```
+
+### Getting Started
+
+We have a few flexible options to get you started with the client. It's meant to cover the majority of common use cases with a simple encapsulation.
+
+#### HTTP/Multipart Usage
+
+Here's a simple example. In this case, a websocket client isn't even initialized, only http (with multipart `@stream` and `@defer` support of course!).
+
+```ts
+import * as React from 'react';
+import ReactDOM from 'react-dom';
+import { GraphiQL } from 'graphiql';
+import { buildGraphiQLFetcher } from '@graphiql/build-fetcher';
+
+const url = 'https://myschema.com/graphql';
+
+const fetcher = buildGraphiQLFetcher({ url });
+
+export const App = () => <GraphiQL fetcher={fetcher} />;
+
+ReactDOM.render(document.getElementByID('graphiql'), <App />);
+```
+
+#### HTTP/Multipart & Websockets
+
+Just by providing the `subscriptionsUrl`, you can generate a `graphql-ws` client
+
+```ts
+import * as React from 'react';
+import ReactDOM from 'react-dom';
+import { GraphiQL } from 'graphiql';
+import { buildGraphiQLFetcher } from '@graphiql/build-fetcher';
+
+const url = 'https://myschema.com/graphql';
+
+const subscriptionsUrl = 'wss://myschema.com/graphql';
+
+const fetcher = buildGraphiQLFetcher({
+  url,
+  subscriptionsUrl,
+});
+
+export const App = () => <GraphiQL fetcher={fetcher} />;
+
+ReactDOM.render(document.getElementByID('graphiql'), <App />);
+```
+
+You can further customize the `wsClient` implementation below
+
+### Options
+
+#### `url` (_required_)
+
+This is url used for all `HTTP` requests, and for schema introspection.
+
+#### `subscriptionsUrl`
+
+This generates a `graphql-ws` client.
+
+#### `wsClient`
+
+provide your own subscriptions client. bypasses `subscriptionsUrl`. In theory, this could be any client using any transport, as long as it matches `graphql-ws` `Client` signature.
+
+#### `headers`
+
+Pass headers to any and all requests
+
+#### `fetch`
+
+Pass a custom fetch implementation such as `isomorphic-feth`
+
+### Customization Examples
+
+#### Custom `wsClient` Example
+
+Just by providing the `subscriptionsUrl`
+
+```ts
+import * as React from 'react';
+import ReactDOM from 'react-dom';
+import { GraphiQL } from 'graphiql';
+import { createClient } from 'graphql-ws';
+import { buildGraphiQLFetcher } from '@graphiql/build-fetcher';
+
+const url = 'https://myschema.com/graphql';
+
+const subscriptionsUrl = 'wss://myschema.com/graphql';
+
+const fetcher = buildGraphiQLFetcher({
+  url,
+  wsClient: createClient({
+    url: subscriptionsUrl,
+    keepAlive: 2000,
```

Comment:
I think that the `keepAlive` is superfluous here. Its mainly useful in applications which have a lot of moving parts where you dont want to waste resources (a lot of connects/disonnects to a WS server).

I would, however, recommend using the `lazy: false` option. This way GraphiQL would immediately connect to a server and the user can know right away if WS works (instead of being forced to issue a subscription operation just to check the basics).

[Something like what PostGraphile's PostGraphiQL does: shows a status badge on the top right indicating the state of the WS connection.](https://github.com/graphile/postgraphile/blob/af39c3dcea40646c9217785fa93a9601612c2b83/postgraphiql/src/components/PostGraphiQL.js#L666-L725)

```suggestion
```

Reply:
oh yeah i was just adding an arbitrary setting so people could see that more than just URL is available. `lazy: false` sounds like a good idea, I can add that to the default implementation